Makefile

```diff
@@ -0,0 +1,37 @@
+PROFILE=release
```

Comment:
I liked that we could previously generate everything with a `cargo build`. Is there a specific reason for switching to a Makefile?

Reply:
See discussion [here](https://github.com/clap-rs/clap/discussions/3848) about generating completions in the build.rs file. 

Basically, getting this to work with generating the completions from build.rs was getting complicated. So I needed a way to do something after the build finished. There's probably a cargo plugin that could do this (for example https://github.com/phil-opp/cargo-post), but a makefile seemed like the most straighforward way to do it, and probably wouldn't require installing anything new.